CspSubmissionPlugin.inc.php

```diff
@@ -904,84 +844,70 @@ function TemplateManager_fetch($hookName, $args) {
 				return true;
 			}
 
-		}elseif ($args[1] == 'controllers/grid/queries/form/queryForm.tpl' && $stageId == "1") {
-
-			import('lib.pkp.classes.mail.MailTemplate');
-			$mail = new MailTemplate('PRE_AVALIACAO');
-			$templateSubject['PRE_AVALIACAO'] = $mail->_data["subject"];
-			$templateBody['PRE_AVALIACAO'] = $mail->_data["body"];
-
-			$templateMgr = TemplateManager::getManager($request);
-			$templateMgr->assign(array(
-				'templates' => $templateSubject,
-				'stageId' => $stageId,
-				'submissionId' => $submissionId,
-				'message' => json_encode($templateBody),
-				'comment' => reset($templateBody)
-			));
+		}elseif ($args[1] == 'controllers/grid/queries/form/queryForm.tpl') {
 
-			$args[4] = $templateMgr->fetch($this->getTemplateResource('queryForm.tpl'));
+			/// Se o usuário for o autor, o destinatário será a secretaria, caso contrário, o destinatário será o autor
+			$submissionDAO = Application::getSubmissionDAO();
+			$submission = $submissionDAO->getById($request->getUserVar('submissionId'));
+			$publication = $submission->getCurrentPublication();
+			$userDao = DAORegistry::getDAO('UserDAO');
+			$authorDao = DAORegistry::getDAO('AuthorDAO');
+			$author = $authorDao->getById($publication->getData('primaryContactId'));
+			$author = $userDao->getUserByEmail($author->getData('email'));
 
-			return true;
+			if ($author->getData('id') == $_SESSION["userId"]) {
 
-		}elseif ($args[1] == 'controllers/grid/queries/form/queryForm.tpl' && $stageId == "4") {
+				$mail = new MailTemplate('MSG_AUTOR_SECRETARIA');
+				$templateSubject['MSG_AUTOR_SECRETARIA'] = $mail->_data["subject"];
+				$templateBody['MSG_AUTOR_SECRETARIA'] = $mail->_data["body"];
 
-			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
-			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
-			$manager = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_MANAGER);
-			$assistent = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_ASSISTANT);
-			$stageAssignmentsFactory = $stageAssignmentDao->getBySubmissionAndStageId($request->getUserVar('submissionId'), null, null, $_SESSION["userId"]);
+				$templateMgr->assign('author', true);
+				$templateMgr->assign('to', 14); // BUSCAR ID DA SECRETARIA
+				$templateMgr->assign('toName', 'Secretaria');
+				$templateMgr->assign('from', $_SESSION["userId"]);
```

Comment:
essas variaveis vão ser usadas no template né? talvez alterar o nome delas pra ficar um pouco mais claro que é id e uma flag, por exemplo
```suggestion
				$templateMgr->assign('isAuthor', true);
				$templateMgr->assign('toId', 14); // BUSCAR ID DA SECRETARIA
				$templateMgr->assign('toName', 'Secretaria');
				$templateMgr->assign('fromId', $_SESSION["userId"]);
```
ou algo assim

Reply:
Sim, está sendo usado no template. O from é o remetente da mensagem. Poderia ser fromId? E o isAuthor, tem alguma sugestão?

Reply:
pode, minhas sugestões são essas ali do bloco, mas o ponto é tentar deixar claro que não é um email, string, e sim um numero ou booleano algo assim